Sources/Build/Destination.swift

```diff
@@ -0,0 +1,128 @@
+            }
+
+            let platformPath = try Process.checkNonZeroExit(args: "xcrun", "--sdk", sdkPath, "--show-sdk-platform-path").chomp()
+            guard !platformPath.isEmpty else {
+                throw DestinationError.invalidInstallation(problem: "could not find SDK platform path for SDK \(sdkPath)")
+            }
+
+            let sdkPlatformFrameworksPath = AbsolutePath(platformPath).appending(components: "Developer", "Library", "Frameworks")
+
+            let devDirPath = try Process.checkNonZeroExit(args: "xcode-select", "-p").chomp()
+            let platformFrameworkPath = AbsolutePath(sdkPath).appending(component: "System")
+                                                             .appending(component: "Library")
+                                                             .appending(component: "Frameworks").asString
+            let toolchainPath = AbsolutePath(devDirPath).appending(component: "Toolchains")
+                                                        .appending(component: "XcodeDefault.xctoolchain")
+            return Destination(target: "x86_64-apple-macosx10.10",
+                               sdk: AbsolutePath(sdkPath),
+                               toolchain: toolchainPath,
+                               dynamicLibraryExtension: "dylib",
+                               extraCCFlags: ["-F", sdkPlatformFrameworksPath.asString],
+                               extraSwiftCFlags: ["-F", platformFrameworkPath,
+                                                  "-F", sdkPlatformFrameworksPath.asString,
+                                                  "-target", "x86_64-apple-macosx10.10",
+                                                  "-sdk", sdkPath],
+                               extraCPPFlags: ["-lc++"])
+        #elseif os(Linux)
+            return Destination(target: "linux-unknown-x86_64",
+                               sdk: AbsolutePath("/"),
+                               toolchain: AbsolutePath(CommandLine.arguments[0],
+                                                       relativeTo: currentWorkingDirectory)
+                                          .parentDirectory.parentDirectory.parentDirectory,
+                               dynamicLibraryExtension: "so",
+                               extraCCFlags: ["-fPIC"],
+                               extraSwiftCFlags: [],
+                               extraCPPFlags: ["-lstdc++"])
+        #else
+            fatalError("unsupported OS, sorry")
+        #endif
+    }
+
+    /// Load a Destination description from a JSON representation from disk.
+    public static func loadFromDescription(path: AbsolutePath,
+                                           fileSystem: FileSystem = localFileSystem) throws -> Destination {
+        let json = try JSON(bytes: try fileSystem.readFileContents(path))
+        return try Destination.init(json: json)
+    }
+
+#if os(macOS)
+    public func sdkPlatformFrameworkPath() -> AbsolutePath? {
```

Comment:
Ah right. Lets make this a property then because we already computed this above.